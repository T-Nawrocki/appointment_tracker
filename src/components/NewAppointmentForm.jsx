import React, { useState } from 'react';

const NewAppointmentForm = (props) => {
  return ( 
    <form id="new-appointment-form">
      
    </form>
   );
}
 
export default NewAppointmentForm;
