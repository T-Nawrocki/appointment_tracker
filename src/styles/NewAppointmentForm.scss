@use 'common/colours';

#input-warnings {
  margin: 2% 25% 2% 25%;
  border: 3px colours.$red solid;
  border-radius: 25px;

  li {
    margin: 2% 0 2% 0;
    color: colours.$red;
    font-weight: bold;
  }
}
