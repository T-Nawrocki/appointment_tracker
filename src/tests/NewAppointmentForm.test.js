import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import NewAppointmentForm from '../components/NewAppointmentForm';

configure({ adapter: new Adapter() });

describe('New Appointment Form', () => {
  it('should render the form', () => {
    const tree = renderer.create(<NewAppointmentForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render the title input', () => {
    const wrapper = shallow(<NewAppointmentForm />);
    expect(wrapper.find('input[name="new-appointment-title"]').exists()).toBe(true);
  });

  it('should render the date input', () => {
    const wrapper = shallow(<NewAppointmentForm />);
    expect(wrapper.find('input[name="new-appointment-date"]').exists()).toBe(true);
  });

  it('should render the time select', () => {
    const wrapper = shallow(<NewAppointmentForm />);
    expect(wrapper.find('select[name="new-appointment-time"]').exists()).toBe(true);
  });

  it('should render the submit input', () => {
    const wrapper = shallow(<NewAppointmentForm />);
    expect(wrapper.find('input[name="new-appointment-submit"]').exists()).toBe(true);
  });
});
