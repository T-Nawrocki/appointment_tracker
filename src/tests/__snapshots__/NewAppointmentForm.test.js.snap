// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Appointment Form should render the form 1`] = `
<form
  id="new-appointment-form"
/>
`;
